### ISSUE TITLE
fix audio not working on ios and safari

### ISSUE THREAD

**Issue by samuelbeard** [charisma-ai/charisma-sdk-js#58]
- [ ] Background audio can play on iOS Safari and other iOS browsers.
- [ ] Background audio ducks (or mutes) when microphone is recording.

It seems that Apple mobile devices don't allow you to set volume in Javascript. We'll need to have a fallback on those devices to mute instead of set volume.

Safari Desktop
- Cannot play audio unless it is manually started by a user interaction.
- Once any audio has been triggered, the page will allow audio on that page.
I've got around this my playing a silent bit of audio at the start of a playthrough that allows the rest of the audio to play.

For iOS though, this doesn't work. The interaction has to be right before the audio starts playing and doesn't persist the life of the page. I can get the background audio to play if I pressed a button just before the media is supposed to play every time.
It doesn't seem to matter after that what interaction it is just as long as there is one. And it's within a couple of seconds before .play() is called.

In summary. I don't think its possible to have background audio on iPhones unless we take a whole new approach to the TrackManager. Character audio plays fine and that uses AudioContext. So I'm going to look at how I can rewrite the track manager to use that instead of HTMLAudioElements.
